aten/src/ATen/native/NativeFunctions.h

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include "ATen/ATen.h"
+#include <vector>
+
+/**
+ * ATen native functions are ways to write ATen methods which make only
+ * make use of other ATen operations (e.g., it is not necessary to
+ * bind into TH/THC code or drop into CUDA kernels.)  These functions
+ * are written as both functions as well as cwrap fragments, which are
+ * then folded into the ATen code generation process; define a function
+ * here, and it will show up as a method on at::Tensor.
+ *
+ * At the moment, only type_method_definition_level: base is supported.
```

Comment:
While we're in the business of parsing the actual C prototypes, there should be a comment to that effect here; if we're going to keep the code like this longer term, there should be some commentary about the limitations of the parser, and where you should look if you need to extend it to support something else.

Reply:
I think a comment that we parse them and where to look is good; any detailed comment about limitations gets out of date the too quickly; I'll write something about general limitations.

Reply:
I'd rather go to the other direction really -- a Yaml string that is parsed and turned into a valid C++ declaration (not definition) rather than the other way around. 

For instance, I'm scared what happens if someone clang-formats this file and everything mysteriously breaks.

Reply:
Yes, my comment here is mostly conditional on, "If we accept this PR as is", even if we plan short term work to make it stop being the case.

Reply:
Let's discuss this below -- but yes, this was always meant as a short term solution to get RoIPooling in and move us closer to the yaml solution.